File: tutorials/python/tutorial_photolithography.py
from semiconductor_simulator.simulator import Simulator

print("Tutorial: Photolithography Module")
simulator = Simulator()
print("Step 1: Initialize grid")
simulator.initialize_geometry(10, 10)
print("Step 2: Run exposure")
mask = [[1, 0, 1], [0, 1, 0], [1, 0, 1]]
simulator.run_exposure(13.5, 0.33, mask)
print("Step 3: Visualize pattern")
simulator.initialize_renderer(400, 400)
simulator.render(simulator.get_wafer())

File: tutorials/cpp/tutorial_photolithography.cpp
#include "../../src/cpp/core/wafer.hpp"
#include "../../src/cpp/modules/geometry/geometry_manager.hpp"
#include "../../src/cpp/modules/photolithography/lithography_model.hpp"
#include <memory>
#include <iostream>

int main() {
  std::cout << "Tutorial: Photolithography Module\n";
  auto wafer = std::make_shared<Wafer>(300.0, 775.0, "silicon");
  GeometryManager geometry;
  std::cout << "Step 1: Initialize grid\n";
  geometry.initializeGrid(wafer, 10, 10);
  LithographyModel lithography;
  std::cout << "Step 2: Run exposure\n";
  std::vector<std::vector<int>> mask = {{1, 0, 1}, {0, 1, 0}, {1, 0, 1}};
  lithography.simulateExposure(wafer, 13.5, 0.33, mask);
  std::cout << "Pattern density: " << wafer->getPhotoresistPattern().sum() / (10 * 10) << "\n";
  return 0;
}